### PR TITLE
[AArch64] Handle ANY_EXTEND in BuildShuffleExtendCombine

### PR DESCRIPTION
**Pull request by hazzlim**
Handle ANY_EXTEND when combining a buildvector/shuffle of extended
operands, as we can safely ignore ANY_EXTENDS when checking if all signs
of the other extends are matching.

### CODE REVIEW

**Review by davemgreen — APPROVED** [review]
I think this looks OK. LGTM

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 1]
```diff
@@ -18594,7 +18608,10 @@ static SDValue performBuildShuffleExtendCombine(SDValue BV, SelectionDAG &DAG) {
                                    : BV.getOperand(1).getOperand(0),
                                cast<ShuffleVectorSDNode>(BV)->getMask());
   }
-  return DAG.getNode(IsSExt ? ISD::SIGN_EXTEND : ISD::ZERO_EXTEND, DL, VT, NBV);
+  unsigned ExtOpc = !SeenZExtOrSExt ? ISD::ANY_EXTEND
+                    : IsSExt        ? ISD::SIGN_EXTEND
+                                    : ISD::ZERO_EXTEND;
```
Comment (NickGuy-Arm) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:18613]: Minor nitpick: I'm not keen on the readability of nested ternary operators like this. Can we add some brackets to make it clear
```suggestion
  unsigned ExtOpc = !SeenZExtOrSExt ? ISD::ANY_EXTEND
                    :(IsSExt        ? ISD::SIGN_EXTEND
                                    : ISD::ZERO_EXTEND);
```

**Review by NickGuy-Arm — APPROVED** [review]
LGTM, barring one minor code formatting nit.

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp (hunk 1 above)
Comment (hazzlim) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.cpp]: Ah yep good point - have added the brackets to improve readability (plus the re-formatting from clang-format).

Thank you for reviewing!

**Pull request merged**